### ISSUE TITLE
Attempt to decode JSON with unexpected mimetype: text/plain

### ISSUE THREAD

**Issue by IMakeTP** [iMicknl/ha-nest-protect#326]
### The problem

Unknown pynest exception.

### What version of this integration (ha-nest-protect) has the issue?

0.4.0b4

### What version of Home Assistant Core has the issue?

2024.4.3

### Device / Model

Nest Protect Battery

### Home Assistant log

<details><summary>Logs</summary>

This error originated from a custom integration.

Logger: custom_components.nest_protect
Source: custom_components/nest_protect/__init__.py:168
integration: Nest Protect (documentation, issues)
First occurred: 6:51:07 PM (4 occurrences)
Last logged: 9:54:12 PM

Unknown pynest exception. Please create an issue on GitHub with your logfile. Updates paused for 1 minute.
Traceback (most recent call last):
  File "/config/custom_components/nest_protect/pynest/client.py", line 298, in subscribe_for_data
    result = await response.json()
             ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/aiohttp_client.py", line 73, in json
    return await super().json(*args, loads=loads, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/aiohttp/client_reqrep.py", line 1176, in json
    raise ContentTypeError(
aiohttp.client_exceptions.ContentTypeError: 0, message='Attempt to decode JSON with unexpected mimetype: text/plain', url=URL('https://czfe73-front01-iad01.transport.home.nest.com/v6/subscribe')

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/config/custom_components/nest_protect/__init__.py", line 168, in _async_subscribe_for_data
    result = await entry_data.client.subscribe_for_data(
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/nest_protect/pynest/client.py", line 302, in subscribe_for_data
    raise PynestException(
custom_components.nest_protect.pynest.exceptions.PynestException: 200 error while subscribing - 

</details>

**Comment by martymart8864** [comment]
Having the same issue and device/versions as @IMakeTP

**Comment by bmos** [comment]
I have this issue also

<details>

<summary>Logs</summary>

Logger: custom_components.nest_protect
Source: custom_components/nest_protect/__init__.py:168
integration: Nest Protect (documentation, issues)
First occurred: 1:00:23 PM (1 occurrences)
Last logged: 1:00:23 PM

Unknown pynest exception. Please create an issue on GitHub with your logfile. Updates paused for 1 minute.
Traceback (most recent call last):
  File "/config/custom_components/nest_protect/pynest/client.py", line 298, in subscribe_for_data
    result = await response.json()
             ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/aiohttp_client.py", line 73, in json
    return await super().json(*args, loads=loads, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/aiohttp/client_reqrep.py", line 1176, in json
    raise ContentTypeError(
aiohttp.client_exceptions.ContentTypeError: 0, message='Attempt to decode JSON with unexpected mimetype: text/plain', url=URL('https://czfe74-front01-iad01.transport.home.nest.com/v6/subscribe')

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/config/custom_components/nest_protect/__init__.py", line 168, in _async_subscribe_for_data
    result = await entry_data.client.subscribe_for_data(
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/nest_protect/pynest/client.py", line 302, in subscribe_for_data
    raise PynestException(
custom_components.nest_protect.pynest.exceptions.PynestException: 500 error while subscribing - 
</details>

**Comment by iMicknl** [comment]
Please update to v0.4.0b5 and let me know if this issue is still present.

**Comment by martymart8864** [comment]
Hi @iMicknl, thanks for the update but stupid question. How do I upgrade? I uninstalled the v0.4.0b4 package from HA, went to the integration again but it reinstalled v0.4.0b4 and not v0.4.0b5. I'm obviously doing something wrong here.. thanks!

**Comment by iMicknl** [comment]
@martymart8864 have you pressed "Update information"? Selecting "Redownload" should show the latest version.

![image](https://github.com/iMicknl/ha-nest-protect/assets/1424596/50b0a56e-3fd5-4126-84af-f0f9e049e0b7)

**Comment by martymart8864** [comment]
Thanks again @iMicknl, got it now. I'll let it run for a a few days and check if I still get the error!

**Comment by martymart8864** [comment]
Hi @iMicknl 

Still given me an error message **running version 0.4.0b5**

------------------------------------------------------------------

Logger: custom_components.nest_protect
Source: custom_components/nest_protect/__init__.py:169
integration: Nest Protect ([documentation](https://github.com/imicknl/ha-nest-protect), [issues](https://github.com/imicknl/ha-nest-protect/issues))
First occurred: 1:36:35 PM (1 occurrences)
Last logged: 1:36:35 PM

Unknown pynest exception. Please create an issue on GitHub with your logfile. Updates paused for 1 minute.
Traceback (most recent call last):
  File "/config/custom_components/nest_protect/pynest/client.py", line 309, in subscribe_for_data
    result = await response.json()
             ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/aiohttp_client.py", line 79, in json
    return await super().json(*args, loads=loads, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/aiohttp/client_reqrep.py", line 1176, in json
    raise ContentTypeError(
aiohttp.client_exceptions.ContentTypeError: 0, message='Attempt to decode JSON with unexpected mimetype: text/plain', url=URL('https://czfe09-front01-iad01.transport.home.nest.com/v6/subscribe')

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/config/custom_components/nest_protect/__init__.py", line 169, in _async_subscribe_for_data
    result = await entry_data.client.subscribe_for_data(
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/nest_protect/pynest/client.py", line 313, in subscribe_for_data
    raise PynestException(
custom_components.nest_protect.pynest.exceptions.PynestException: 200 error while subscribing -

**Comment by GSzabados** [comment]
I looked at this issue, the code which creates the error is this:

```
        ) as response:
            _LOGGER.debug("Data received via subscriber (status: %s)", response.status)

            if response.status == 401:
                raise NotAuthenticatedException(await response.text())

            if response.status == 504:
                raise GatewayTimeoutException(await response.text())

            if response.status == 502:
                raise BadGatewayException(await response.text())

            try:
                result = await response.json()
            except ContentTypeError as error:
                result = await response.text()
                
                raise PynestException(
                    f"{response.status} error while subscribing - {result}"
                ) from error

            # TODO type object
            return result
```

This `f"{response.status} error while subscribing - {result}"`, and the result in the log `raise PynestException(
custom_components.nest_protect.pynest.exceptions.PynestException: 200 error while subscribing -` suggests to me, that the `response` is empty, hence there are no further text after the `-` as `{result}`, being `result = await response.text()`.

This exception/error is just not handled gracefully, as there is no special if for `status == 200` and `ContentType` `mimetype: text/plain`.

**Comment by iMicknl** [comment]
@GSzabados would you be willing to do a PR (to the beta branch)? :-)

**Comment by GSzabados** [comment]
I would, but I am not a Python expert. I am using ChatGPT to try to figure out what is going on.

But likely I can fix this one. And make a PR.

Are you alright with my logic there? Have I missed something?

**Comment by GSzabados** [comment]
> @GSzabados would you be willing to do a PR (to the beta branch)? :-)

I did the PR, that should fix this issue and #345 as well, because they are duplicates.
